lib/ansible/modules/windows/win_disk_facts.ps1

```diff
@@ -0,0 +1,181 @@
+#!powershell
+
+# Copyright: (c) 2017, Marc Tschapek <marc.tschapek@itelligence.de>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.Legacy
+$ErrorActionPreference = "Stop"
+Set-StrictMode -Version 2.0
+
+# Create a new result object
+$result = @{
+    changed = $false
+    ansible_facts = @{
+        ansible_disk = @(
+        )
+    }
+}
+
+# Search disks
+if (Get-Command 'Get-Disk') {
+    try {
+        $disks = Get-Disk
+    } catch {
+        Fail-Json -obj $result -message "Failed to search the disks on the target: $($_.Exception.Message)"
+    }
+    if ($disks) {
+        [int32]$diskcount = $disks | Measure-Object | Select-Object  -ExpandProperty Count
+        $result.ansible_facts.ansible_disk += @{total_disks_found = $diskcount}
+        foreach ($disk in $disks) {
+            $disk_results = @{}
+            $pdisk = Get-PhysicalDisk -ErrorAction SilentlyContinue | Where-Object {
+                $_.DeviceId -eq $disk.Number
+            }
+            if ($pdisk) {
+                $disk_results["physical_disk"] += @{
+                    #media_type = $pdisk.MediaType
+                    size = "$($pSize = "{0:N3}" -f ($pdisk.Size / 1GB))$($pSize)gb"
```

Comment:
I'm wondering whether this should just the size in bytes without any unit prefix to make things simple. The benefit of this approach is that you have a raw integer in the variable key that people can now easily perform operations on. The tradeoff is GB and TB values will be quite large, what are your thoughts?

Reply:
In my opinion we should use the formatted approach (rounded and with prefix) if the module is only used to display the disk facts. But if it the values will be used for further calculations or another module we should remove the prefix and rounding.
What do you think?